modules/tinymce/src/plugins/table/main/ts/queries/TableSize.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { HTMLTableElement } from '@ephox/dom-globals';
+import { TableSize } from '@ephox/snooker';
+import { Element, Traverse, Width } from '@ephox/sugar';
+import Editor from 'tinymce/core/api/Editor';
+import { calculatePercentageWidth } from '../actions/EnforceUnit';
+import { isPercentagesForced, isPixelsForced } from '../api/Settings';
+import * as Util from '../core/Util';
+
+export const get = (editor: Editor, table: Element<HTMLTableElement>) => {
+  // Note: We can't enforce none (responsive), as if someone manually resizes a table
+  // then it must switch to either pixel (fixed) or percentage (relative) sizing
+  if (isPercentagesForced(editor)) {
+    return Util.getRawWidth(editor, table.dom())
```

Comment:
Maybe extract a `forcePercentages` function. Then this whole `if` block will read really nicely.

Reply:
A lot of this is changing with the following TINY-6000 PR, so this will change then. We also don't want to force the percentage at this point, as it's only meant to be a lookup.